lib/explorer/data_frame.ex

```diff
@@ -2799,7 +2799,7 @@ defmodule Explorer.DataFrame do
           datetime = %NaiveDateTime{} ->
             lazy_s = LazySeries.new(:to_lazy, [datetime])
 
-            Explorer.Backend.Series.new(lazy_s, :datetime)
+            Explorer.Backend.Series.new(lazy_s, {:datetime, :micro_seconds})
```

Comment:
Maybe we should keep the default of `microsecond`, which is the precision Elixir uses for its own datetime. WDYT?